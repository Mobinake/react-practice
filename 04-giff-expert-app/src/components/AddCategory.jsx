/* eslint-disable react/prop-types */
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();
		if (inputValue.trim().length <= 1) return;
		setInputValue('');
		onNewCategory(inputValue.trim());
	};

	return (
		<form
			onSubmit={onSubmit}
			aria-label='form'>
			<input
				type='text'
				placeholder='BuscarGifs'
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};

AddCategory.protoTypes = { onNewCategory: PropTypes.func.isRequired };
