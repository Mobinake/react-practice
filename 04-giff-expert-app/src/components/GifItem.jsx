/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';

export const GifItem = ({ title, id, url }) => {
	return (
		<div className="card">
			<img
				src={url}
				alt={title}
			/>
			<li key={id}>{title}</li>;
		</div>
	);
};

GifItem.PropTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
