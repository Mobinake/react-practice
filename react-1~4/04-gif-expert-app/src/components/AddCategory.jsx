import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState();

	// cada vez que haya un cambio, obtenerlo
	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	//cada vez que se haga submit(enter), enviar
	const onSubmit = (event) => {
		//prevenir el default de recargar la pagina
		event.preventDefault();

		//quitar espacio en blanco al inicio y final
		if (inputValue.trim().length <= 1) return;

		//setear la categoria despues de los valores ya existentes
		// setCategories((categories) => [inputValue, ...categories]);
		setInputValue('');
		onNewCategory(inputValue.trim());
	};

	return (
		//un formulario que cuando se haga submit, vaya a la funicon onSubmit
		<form onSubmit={onSubmit}>
			<input
				types="text"
				placeholder="Buscar gifs"
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};
