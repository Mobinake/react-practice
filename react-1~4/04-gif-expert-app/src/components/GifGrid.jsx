import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { images, isLoading } = useFetchGifs(category);

	console.log({ images, isLoading });

	return (
		<>
			<h4>{category}</h4>
			{isLoading && <h2>Cargando...</h2>}
			<div className="card-grid">
				{images.map((image) => (
					<GifGridItem
						key={image.id}
						//esparcir las propiedades, enviarlas todas
						{...image}
					/>
				))}
			</div>
		</>
	);
};
