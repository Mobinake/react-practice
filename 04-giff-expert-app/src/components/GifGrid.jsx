/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { images, isLoading } = useFetchGifs(category);

	console.log(images, isLoading);

	return (
		<>
			<h4>{category}</h4>
			{isLoading && <h3>Cargando......</h3>}

			<div className='card-grid'>
				{images.map((image) => (
					<GifItem
						key={image.id}
						{...image}
					/>
				))}
			</div>
		</>
	);
};
