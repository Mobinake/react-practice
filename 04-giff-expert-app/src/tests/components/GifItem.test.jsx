import { render } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Este modulo hace testing de <GifItem/>', () => {
	const title = 'TÃ­tulo de prueba';
	const url = 'https://media.giphy.com/media/example/giphy.gif';

	test('debe hacer match con el snapshot', () => {
		const { container } = render(
			<GifItem
				title={title}
				url={url}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
