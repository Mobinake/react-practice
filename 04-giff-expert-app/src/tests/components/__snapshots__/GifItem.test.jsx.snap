// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Este modulo hace testing de <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título de prueba"
      src="https://media.giphy.com/media/example/giphy.gif"
    />
    <li>
      Título de prueba
    </li>
    ;
  </div>
</div>
`;
